flexget/plugins/input/dynamic_imdb.py

```diff
@@ -112,130 +138,126 @@ def get_entity_type_and_object(self, imdb_id):
             log.info('Starting to retrieve items for Character: %s' % character)
             return 'Character', character
 
-    def get_items_by_person(self, person, job_types, content_types):
+    def get_items_by_entity(self, entity_type, entity_object, content_types, job_types):
         """
-        Return item list for a person object
-        :param person: Person object
-        :param job_types: Relevant job types as inputted in config
-        :param content_types: Relevant content types as inputted in config
-        :return: Person item list
+        Gets entity object and return movie list using relevant method
         """
-        items = []
-        for job_type in job_types:
-            if job_type in JOBS_WITHOUT_CONTENT_TYPE:
-                log.debug('Searching for movies that correlates to: ' + job_type)
-                items += person.get(job_type)
-                continue
+        if entity_type == 'Company':
+            return self.items_by_company(entity_object)
+
+        if entity_type == 'Character':
+            return self.items_by_character(entity_object, content_types)
+
+        elif entity_type == 'Person':
+            return self.items_by_person(entity_object, job_types, content_types)
+
+    def flatten_list(self, _list):
+        """
+        Gets a list of lists and returns a flat list
+        """
+        for el in _list:
+            if isinstance(el, collections.Iterable) and not isinstance(el, basestring):
+                for sub in self.flatten_list(el):
+                    yield sub
             else:
-                for content_type in content_types:
-                    job_and_content = job_type + ' ' + CONTENT_TYPE_CONVERSION[content_type]
-                    log.debug('Searching for movies that correlates to: ' + job_and_content)
-                    movies_by_job_type = person.get(job_and_content)
-                    if content_type == 'movie':
-                        log.debug('Searching for short movies and documentaries.')
-                        movies_by_job_type += person.get(job_type + ' ' + 'documentary', [])
-                        movies_by_job_type += person.get(job_type + ' ' + 'short', [])
-                    if movies_by_job_type:
-                        for item in movies_by_job_type:
-                            self.ia.update(item)
-                            if item not in items and item['kind'] in content_types:
-                                log.debug('Found item: ' + item.get('title') + ', adding to unfiltered list')
-                                items.append(item)
-                            else:
-                                log.debug('Item ' + str(item) + ' already found in list, skipping.')
-        return items
+                yield el
 
-    def get_items_by_character(self, character, content_types):
+    def no_empty_objects(self, _list):
```

Comment:
There's a builtin for this, `filter(None, thelist)` (the first argument is a discriminator function, but if you pass None it just filters items that aren't truthy.)